include/fem-dem/ib_particles_dem.h

```diff
@@ -0,0 +1,182 @@
+//
+// Created by lucka on 2021-10-11.
+//
+
+#ifndef LETHE_IB_PARTICLES_DEM_H
+#define LETHE_IB_PARTICLES_DEM_H
+
+#include <core/ib_particle.h>
+#include <core/ib_stencil.h>
+#include <core/lethegridtools.h>
+
+#include <solvers/navier_stokes_base.h>
+
+#include <deal.II/base/tensor.h>
+
+#include <deal.II/dofs/dof_handler.h>
+#include <deal.II/dofs/dof_tools.h>
+
+using namespace dealii;
+using namespace std;
+
+
+/**
+ * A solver class for the DEM used in conjunction with IB particles and
+ * gls_sharp_navier_stokes. This class defines and use some function of the DEM
+ * class that has been modified and simplified to use IB_particles.
+ *
+ * @tparam dim An integer that denotes the dimension of the space in which
+ * the flow is solved
+ *
+ * @ingroup solvers
+ * @author Lucka Barbeau, Bruno Blais, Shahab Golshan 2021
+ */
+
+template <int dim>
+class IBParticlesDEM
+{
+public:
+  /**
+   * @brief
+   * Initialize the IBParticlesDEM object with the parameter, the mpi
+   * communicator and the particles.
+   *
+   * @param p_nsparam The parameter of the simulation.
+   *
+   * @param mpi_communicator_input The mpi communicator of the simulation.
+   *
+   * @param particles The particles vector containing all the IB particles.
+   */
+  void
+  initialize(SimulationParameters<dim>    p_nsparam,
```

Comment:
```suggestion
  initialize(SimulationParameters<dim>    p_nsparam,
```
Why do you need to pass the entire simulation parameters? Just pass the section you need. Otherwise this makes testing these functions very complicated.